core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -41,6 +41,9 @@ private[spark] abstract class MemoryManager(
     onHeapStorageMemory: Long,
     onHeapExecutionMemory: Long) extends Logging {
 
+  require(onHeapStorageMemory > 0, "onHeapStorageMemory must be > 0")
```

Comment:
Hm, on second thought, there's not really a reason this must be positive. It's at least possible that no memory is immune from eviction.